File: solutions/D. 히스토그램.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        int n = scanner.nextInt();
        int[] array = new int[n];
        
        int maxHeight = 0;
        for (int index = 0; index < n; index++) {
            array[index] = scanner.nextInt();
            maxHeight = Math.max(maxHeight, array[index]);
        }

        for (int height = maxHeight; height >= 1; height--) {
            for (int index = 0; index < n; index++) {
                String output;
                if (array[index] >= height) {
                    output = String.valueOf(array[index]);
                } else {
                    output = " ";
                }
                sb.append(output);
            }
            sb.append("\n");
        }
        System.out.println(sb);
        scanner.close();
    }
}

File: solutions/D. 히스토그램.cpp
#include<iostream>
#include<vector>

using namespace std;

int N;
vector<int> Array;

void solve() {
    int max_height = 0;
    for (int index = 0; index < N; index++) {
        cin >> Array[index];
        max_height = max(max_height, Array[index]);
    }

    for (int height = max_height; height >= 1; height--) {
        for (int index = 0; index < N; index++) {
            char output;
            if (Array[index] >= height) {
                output = (char) (Array[index] + '0');
            } else {
                output = ' ';
            }
            cout << output;
        }
        cout << endl;
    }
}

int main() {
    ios::sync_with_stdio(false), cin.tie(nullptr);
    cin >> N;
    Array.resize(N);
    solve();
    return 0;
}
